provider/team_resource.go

```diff
@@ -33,6 +33,26 @@ func resourceTeam() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			"memberships": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"incident_role_id": {
```

Comment:
I was a bit confused about why I was adding an incident role to the Team, but it makes sense after looking at it in the product.  Should this be something like `default_incident_role_id`?

Reply:
possibly, I was keeping it one to one with the API but we can for sure have terraform be the translation layer

Reply:
Interesting -- it comes back in the payload as a "default incident role" so I'd have thought we'd do it the same way on the POST.

That makes it a bit trickier, but I think I'd opt for the usability here knowing that the APIs will continue to evolve under us anyway.